src/nvim/eval.c

```diff
@@ -14880,7 +14880,9 @@ static void f_setreg(typval_T *argvars, typval_T *rettv, FunPtr fptr)
   rettv->vval.v_number = 0;
 
   if (set_unnamed) {
-    op_register_set_previous(regname);
+    if (!op_register_set_previous(regname)) {
+      EMSG("Unable to set previous yank register");
```

Comment:
Missing error number. Need to be something like `E5xxx`, one of the unoccupied. And I would suggest using `emsgf("E5542: Unable to set previous yank register to '%c' (<%02x>)", regname, (unsigned)regname)`: as far as I understand `regname` is the only possible reason of the inabliity so this variant will provide all needed details. (Though `(<%02x>)` is probably not needed, but I do not remember how different non-printable characters will work there.)